browse/services/documents/db_implementation/db_abs.py

```diff
@@ -0,0 +1,140 @@
+"""Legacy DB backed core metadata service."""
+from typing import Dict, List, Optional
+from dataclasses import replace
+
+from browse.domain.metadata import DocMetadata
+from browse.domain.identifier import Identifier
+from browse.services.documents.config.deleted_papers import DELETED_PAPERS
+
+from browse.services.documents.base_documents import DocMetadataService, \
+    AbsDeletedException, AbsNotFoundException , AbsVersionNotFoundException
+
+from browse.services.database.models import Metadata
+
+from ..format_codes import formats_from_source_type
+from .convert import to_docmeta
+
+
+class DbDocMetadataService(DocMetadataService):
+    """Class for arXiv document metadata service."""
+
+    
+    def __init__(self,
+                 db) -> None:
+        """Initialize the DB document metadata service."""
+        self.db = db
+
+    
+    def get_abs(self, arxiv_id: str) -> DocMetadata:
+        """Get the .abs metadata for the specified arXiv paper identifier.
+
+        Parameters
+        ----------
+        arxiv_id : str
+            The arXiv identifier string.
+
+        Returns
+        -------
+        :class:`DocMetadata`
+        """
+
+        # TODO Probably doesn't do docmeta.version_history correctly
+        paper_id = Identifier(arxiv_id=arxiv_id)
+        if paper_id.id in DELETED_PAPERS:
+            raise AbsDeletedException(DELETED_PAPERS[paper_id.id])
```

Comment:
This will need attention, per TODO

Reply:
I added this to the description of https://arxiv-org.atlassian.net/browse/ARXIVNG-4282